### ISSUE TITLE
Can built without NodeJS if xterm.js is prebuilt or load from cdn just like https://github.com/cs01/pyxtermjs

### ISSUE THREAD

**Issue by equiclouds** [nickvdyck/webtty#150]
[pyxtermjs](https://github.com/cs01/pyxtermjs) is a implemention by python, but it only have no more than 200 lines of code.
Can the solution reconsitution to make it more easier to understand?

**Comment by nickvdyck** [comment]
Yeah fair point about it being a bit convoluted, this used to be a personal playground for me to learn about pty's and building a custom real-time messaging protocol. It used to be even more complex when all the pty logic was still in this project, but I won't go there since this is already abstracted in a separate project.  I have been thinking about improving this and issues like this one https://github.com/nickvdyck/webtty/issues/148 will definitely help. So the codebase will become easier over the coming weeks, but getting it down to just 200 lines of code is not going to happen in C# 😄 .

At the moment there is a hard need for NodeJS in order for this project to build. I have no plans to implement it in such a way that loading all deps from a CDN is possible. But I have been thinking about separating concerns so that there are clear API boundaries and a dotnet library that would make this possible with just a few lines of code.
